### PR TITLE
ci: don't cancel other unit tests when one fails

### PR DESCRIPTION
**Pull request by james-garner-canonical**
A test can pass on one python/os version and fail on others, and it’s nice to see that in the outcome, rather than dig into cancelled tests which may or may not have hit the failing case.

Since unit tests are cheap and fast, we’re not really saving much by cancelling them – by the time one unit test run fails and asks the others to cancel, they’ll already have been spun up and run many of the tests, so cancelling them doesn't save much.

The pip-install and test-real-pebble jobs are faster than the test job, so this commit adds fail-fast: false to them too.

### CODE REVIEW

**Review by dimaqq — APPROVED** [review]

**Review by tonyandrewmeyer — APPROVED** [review]

**Review by benhoyt — APPROVED** [review]

**Comment by james-garner-canonical** [comment]
We can wait till after the release to merge this

**Comment by tonyandrewmeyer** [comment]
> We can wait till after the release to merge this

@james-garner-canonical want to push the big green button now? :smile:

**Comment by dimaqq** [comment]
Or now :)

**Pull request merged**
